src/imitation/policies/trainer.py

```diff
@@ -45,6 +46,42 @@ def train(self, steps: int, **kwargs):
                 the training procedure.
         """
 
+    def set_logger(self, custom_logger: Optional[logger.HierarchicalLogger]):
+        """Set the internal logger.
+        For subclasses, this should also set the logger of any wrapped modules
+        (such as SB3 algorithms).
+
+        Args:
+            logger: the HierarchicalLogger instance to use.
+        """
+        self._logger = custom_logger or logger.configure()
+
+
+class TrajectoryDataset(TrajectoryGenerator):
+    def __init__(self, path: types.AnyPath, seed: int = 0):
+        self._trajectories = types.load(path)
+        self.rng = random.Random(seed)
+
+    def sample(self, steps: int) -> Sequence[types.TrajectoryWithRew]:
+        available_steps = sum(len(traj) for traj in self._trajectories)
+        if available_steps < steps:
+            raise RuntimeError(
+                f"Asked for {steps} transitions but only {available_steps} available"
+            )
+        trajectories = list(self._trajectories)
```

Comment:
Similar logic to `AgentTrainer` -- factor out into helper method?

Reply:
Good idea, done